BaseStation/SensorDiagnostics/SensorDiagnostics/MainWindow.xaml.cs

```diff
@@ -0,0 +1,306 @@
+            DirectoryInfo di = new DirectoryInfo(LogFilesLocation);
+            data = di.GetFiles("*.csv");
+        }
+
+        private void makeGraph()
+        {
+
+            xmax = canGraph.Width - margin;
+            ymax = canGraph.Height - margin;
+
+            // Make the X axis.
+            GeometryGroup xaxis_geom = new GeometryGroup();
+            xaxis_geom.Children.Add(new LineGeometry(
+                new Point(0, ymax), new Point(canGraph.Width, ymax)));
+            for (double x = xmin + step;
+                x <= canGraph.Width - step; x += step)
+            {
+                xaxis_geom.Children.Add(new LineGeometry(
+                    new Point(x, ymax - margin / 2),
+                    new Point(x, ymax + margin / 2)));
+            }
+
+            System.Windows.Shapes.Path xaxis_path = new System.Windows.Shapes.Path();
+            xaxis_path.StrokeThickness = 1;
+            xaxis_path.Stroke = Brushes.Black;
+            xaxis_path.Data = xaxis_geom;
+
+            canGraph.Children.Add(xaxis_path);
+
+            // Make the Y ayis.
+            GeometryGroup yaxis_geom = new GeometryGroup();
+            yaxis_geom.Children.Add(new LineGeometry(
+                new Point(xmin, 0), new Point(xmin, canGraph.Height)));
+            for (double y = step; y <= canGraph.Height - step; y += step)
+            {
+                yaxis_geom.Children.Add(new LineGeometry(
+                    new Point(xmin - margin / 2, y),
+                    new Point(xmin + margin / 2, y)));
+            }
+
+            System.Windows.Shapes.Path yaxis_path = new System.Windows.Shapes.Path();
+            yaxis_path.StrokeThickness = 1;
+            yaxis_path.Stroke = Brushes.Black;
+            yaxis_path.Data = yaxis_geom;
+
+            canGraph.Children.Add(yaxis_path);
+        }
+
+        private void makeDropBox()
+        {
+            ComboBox cbox = new ComboBox();
+            cbox.Background = Brushes.LightGray;
+            for (int i = 0; i < data.Length; i++)
+            {
+                ComboBoxItem cboxitem = new ComboBoxItem();
+                cboxitem.Content = data[i].Name;
+                cbox.Items.Add(cboxitem);
+            }
+            cbox.SelectionChanged += (object sender, SelectionChangedEventArgs args) =>
+            {
+                selectedFile = cbox.SelectedIndex;
+                updateGraph();
+            };
+            grid.Children.Add(cbox);
+        }
+
+        private void updateGraph()
+        {
+            StreamReader sr = data[selectedFile].OpenText();
+            while (sr.Peek() >= 0)
+            {
+                int[] sensorNums = new int[7];
+                string[] sensorLogData = sr.ReadLine().Split(Delimiter);
+                PointCollection points = createPoints(sensorLogData);
+                string sensorType = sensorLogData[0];
+                Color brushColor = Color.FromRgb(0, 0, 0);
+                if (encoderRegex.IsMatch(sensorType))
```

Comment:
Can we generalize this so it's not a giant if/else structure?

Reply:
Which part are you referring to?

Reply:
Lines 138-227. Maybe utilize subclasses or a dictionary